### DISCUSSION TITLE
Changing the API Key on the fly

### DISCUSSION THREAD

**Discussion by tjscience** [microsoft/semantic-kernel#8336] [Q&A] [answered]
We allow customers to bring their own API key when using AI services. I see that the API key is required when creating the service:
`builder.Services.AddOpenAIChatCompletion(modelId: "gpt-4o", apiKey: "...");`
Is there a way to dynamically change the key when using the service?

**Comment by sophialagerkranspandey** [comment, marked as answer]
You could set a different instance for each of the keys and select the one you want to use. AI Service selector can help with this as well.
